shell/platform/linux/fl_string_codec_test.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_string_codec.h"
+#include "flutter/shell/platform/linux/fl_test.h"
+#include "gtest/gtest.h"
+
+static gchar* encode_value(FlValue* value) {
+  g_autoptr(FlStringCodec) codec = fl_string_codec_new();
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(GBytes) message =
+      fl_message_codec_encode_message(FL_MESSAGE_CODEC(codec), value, &error);
+  EXPECT_NE(message, nullptr);
+  EXPECT_EQ(error, nullptr);
+
+  return bytes_to_hex_string(message);
+}
+
+static void encode_value_error(FlValue* value, GQuark domain, int code) {
+  g_autoptr(FlStringCodec) codec = fl_string_codec_new();
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(GBytes) message =
+      fl_message_codec_encode_message(FL_MESSAGE_CODEC(codec), value, &error);
+  EXPECT_EQ(message, nullptr);
+  EXPECT_TRUE(g_error_matches(error, domain, code));
+}
+
+static FlValue* decode_value(const char* hex_string) {
+  g_autoptr(FlStringCodec) codec = fl_string_codec_new();
+  g_autoptr(GBytes) message = hex_string_to_bytes(hex_string);
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(FlValue) value =
+      fl_message_codec_decode_message(FL_MESSAGE_CODEC(codec), message, &error);
+  EXPECT_EQ(error, nullptr);
+  EXPECT_NE(value, nullptr);
+  return fl_value_ref(value);
+}
+
+TEST(FlStringCodecTest, EncodeData) {
+  g_autoptr(FlValue) value = fl_value_string_new("hello");
+  g_autofree gchar* hex_string = encode_value(value);
+  EXPECT_STREQ(hex_string, "68656c6c6f");
+}
+
+TEST(FlStringCodecTest, EncodeEmpty) {
+  g_autoptr(FlValue) value = fl_value_string_new("");
+  g_autofree gchar* hex_string = encode_value(value);
+  EXPECT_STREQ(hex_string, "");
+}
+
+TEST(FlStringCodecTest, EncodeNULL) {
+  encode_value_error(nullptr, FL_MESSAGE_CODEC_ERROR,
+                     FL_MESSAGE_CODEC_ERROR_UNSUPPORTED_TYPE);
+}
+
+TEST(FlStringCodecTest, EncodeUnknownType) {
+  g_autoptr(FlValue) value = fl_value_null_new();
+  encode_value_error(value, FL_MESSAGE_CODEC_ERROR,
+                     FL_MESSAGE_CODEC_ERROR_UNSUPPORTED_TYPE);
+}
+
+TEST(FlStringCodecTest, DecodeData) {
+  g_autoptr(FlValue) value = decode_value("68656c6c6f");
+  ASSERT_EQ(fl_value_get_type(value), FL_VALUE_TYPE_STRING);
+  ASSERT_STREQ(fl_value_get_string(value), "hello");
+}
+
+TEST(FlStringCodecTest, DecodeEmpty) {
+  g_autoptr(FlValue) value = decode_value("");
+  ASSERT_EQ(fl_value_get_type(value), FL_VALUE_TYPE_STRING);
+  ASSERT_STREQ(fl_value_get_string(value), "");
+}
```

Comment:
I find it nice in the codec tests to have at least one explicit round-trip test as well.